pkg/controllers/cleanup/controller.go

```diff
@@ -0,0 +1,159 @@
+package cleanup
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-logr/logr"
+	kyvernov1alpha1 "github.com/kyverno/kyverno/api/kyverno/v1alpha1"
+	kyvernov1alpha1informers "github.com/kyverno/kyverno/pkg/client/informers/externalversions/kyverno/v1alpha1"
+	kyvernov1alpha1listers "github.com/kyverno/kyverno/pkg/client/listers/kyverno/v1alpha1"
+	"github.com/kyverno/kyverno/pkg/config"
+	controllerutils "github.com/kyverno/kyverno/pkg/utils/controller"
+	batchv1 "k8s.io/api/batch/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	batchv1informers "k8s.io/client-go/informers/batch/v1"
+	"k8s.io/client-go/kubernetes"
+	batchv1listers "k8s.io/client-go/listers/batch/v1"
+	"k8s.io/client-go/util/workqueue"
+)
+
+type Controller struct {
+	// clients
+	client kubernetes.Interface
+
+	// listers
+	cpolLister kyvernov1alpha1listers.ClusterCleanupPolicyLister
+	polLister  kyvernov1alpha1listers.CleanupPolicyLister
+	cjLister   batchv1listers.CronJobLister
+
+	// queue
+	queue workqueue.RateLimitingInterface
+}
+
+const (
+	maxRetries     = 10
+	Workers        = 3
+	ControllerName = "cleanup-controller"
+)
+
+func NewController(
+	client kubernetes.Interface,
+	cpolInformer kyvernov1alpha1informers.ClusterCleanupPolicyInformer,
+	polInformer kyvernov1alpha1informers.CleanupPolicyInformer,
+	cjInformer batchv1informers.CronJobInformer,
+) *Controller {
+	c := &Controller{
+		client:     client,
+		cpolLister: cpolInformer.Lister(),
+		polLister:  polInformer.Lister(),
+		cjLister:   cjInformer.Lister(),
+		queue:      workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), ControllerName),
+	}
+	controllerutils.AddDefaultEventHandlers(logger, cpolInformer.Informer(), c.queue)
+	controllerutils.AddDefaultEventHandlers(logger, polInformer.Informer(), c.queue)
+	controllerutils.AddEventHandlers(cjInformer.Informer(), c.handleCronJobAddDelete, c.handleCronJobUpdate, c.handleCronJobAddDelete)
+	return c
+}
+
+func (c *Controller) handleCronJobAddDelete(obj interface{}) {
+	cronjob := obj.(batchv1.CronJob)
+	if len(cronjob.OwnerReferences) == 1 {
+		if cronjob.OwnerReferences[0].Kind == "ClusterCleanupPolicy" {
+			c.queue.Add(kyvernov1alpha1.ClusterCleanupPolicy{
```

Comment:
It's not going to work, you need to enqueue keys, not the actual objects.

Reply:
@NikhilSharmaWe please take the code i sent on slack, it will take care of the key encoding.

Reply:
The safest/easiest is to get the enqueue func and reuse that.